groups/etcd_group_service.go

```diff
@@ -189,20 +194,24 @@ func (c *EtcdGroupService) Leave(ctx context.Context, groupName, uid string) err
 	if err != nil {
 		return err
 	}
-	_, err = clientInstance.Delete(ctx, userGroupKey(groupName, uid))
+	_, err = clientInstance.Delete(ctx, memberGroupKey(groupName, uid))
 	return err
 }
 
 // LeaveAll clears all UIDs in the group
 func (c *EtcdGroupService) LeaveAll(ctx context.Context, groupName string) error {
-	dResp, err := clientInstance.Delete(ctx, memberKey(groupName, ""), clientv3.WithPrefix())
+	dResp, err := clientInstance.Delete(ctx, groupPrefix(groupName), clientv3.WithPrefix())
 	if err != nil {
 		return err
 	}
+	prefix := memberKey(groupName, "")
 	for _, kv := range dResp.PrevKvs {
-		_, err = clientInstance.Delete(ctx, string(kv.Key))
-		if err != nil {
-			logger.Log.Warn("[groups] sd: error deleting key from etcd")
+		if strings.Contains(string(kv.Key), prefix) {
+			uid := string(kv.Key)[len(prefix):]
+			_, err = clientInstance.Delete(ctx, memberGroupKey(groupName, uid))
+			if err != nil {
+				logger.Log.Warn("[groups] sd: error deleting key from etcd")
```

Comment:
Log the error here: `logger.Log.Warnf("[groups] sd: error deleting key from etcd: %s", err.Error())`